js/charts/bullet.js

```diff
@@ -526,63 +528,66 @@ function bullet() {
                 .on('mousemove', _handleMouseMoveFn.call(chart, tooltip, _local_svg))
                 .on('mouseout', _handleMouseOutFn.call(chart, tooltip, _local_svg))
                 .on('click', function (d) {
-                    if (isLiveEnabled) {
-                        broadcast.$broadcast('FlairBi:livemode-dialog');
-                        return;
-                    }
-                    var confirm = parentContainer.select('.confirm')
-                        .style('visibility', 'visible');
-
-                    var rect = d3.select(this).select('rect.measure');
+                    if (COMMON.COMPARABLE_DATA_TYPES.indexOf(_dimensionType[0]) === -1) {
 
-                    if (rect.classed('selected')) {
-                        rect.classed('selected', false);
-                        filterData = filterData.filter(function (val) {
-                            if (val[dimension[0]] != d.title) {
-                                return val;
-                            }
-                        })
-                    } else {
-                        rect.classed('selected', true);
-                        var obj = new Object();
-                        obj[dimension] = d.title;
-                        obj[measures[0]] = d.measures.toString();
-                        obj[measures[1]] = d.markers.toString();
+                        if (isLiveEnabled) {
```

Comment:
these is huge function..try to create small-2 function based on functionality...so that i can be easy to read and test.....also there are many lines of code which can be duplicate